### ISSUE TITLE
Add search by Key to the Inspector

### ISSUE THREAD

**Issue by kenzieschmoll** [flutter/devtools#6870]
From user survey: "difficult to walk the tree to find the widget I'm working on, a search function by key would be helpful"

**Comment by ahmadrafidev** [comment]
Hi @kenzieschmoll, I would love to work on this issue. Can you assign it to me and explain the issue to me a bit. Thanks!

**Comment by kenzieschmoll** [comment]
Hi @ahmadrafidev. Thanks for you interest in contributing to this project! 

Right now, the Flutter Inspector supports searching by widget name:
![Screenshot 2024-01-19 at 11 07 50 AM](https://github.com/flutter/devtools/assets/43759233/48c3ebe5-0a10-46df-93d5-88e865d68b26)

This issue is a feature request to not only match by widget name, but to also match by Key. So for a widget:
```
Container(
  key: Key('photo gallery')
  child: ...
)
```
a search for "gallery" would also return this Container widget as a match.

The search match logic for the inspector is here: https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/inspector/inspector_tree_controller.dart/#L658-L659
and here:
https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/inspector/inspector_tree_controller.dart/#L697

**Comment by kenzieschmoll** [comment]
Hi @ahmadrafidev, thank you for looking into this issue. I have good news and bad news. The good news is, after looking at your PR and the before and after videos you included, I realized that searching by a widget key is already supported in the Flutter inspector! In the video you linked, the version of the inspector you were using inside VS code was actually the inspector from the released version of DevTools, not the one you were trying to make local changes to. I confirmed on my side as well that the Flutter Inspector already supports search by key:
![Screenshot 2024-01-22 at 11 11 34 AM](https://github.com/flutter/devtools/assets/43759233/e06ed12b-8349-4e97-8096-c57d2393c8fc)

The bad news is, we'll have to close your PR. But thank you for looking into this because it helped us realize this is already supported, and we can go ahead and close this issue. Thanks!
